cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillRunnerService.java

```diff
@@ -209,15 +209,21 @@ public TwillPreparer prepare(TwillRunnable runnable,
   public TwillPreparer prepare(TwillApplication application) {
     TwillSpecification spec = application.configure();
     RunId runId;
+    String appVersion;
     if (application instanceof ExtendedTwillApplication) {
       runId = RunIds.fromString(((ExtendedTwillApplication) application).getRunId());
+      appVersion = ((ExtendedTwillApplication) application).getApplicationVersion();
     } else {
+      appVersion = null;
```

Comment:
can you add a comment about when we would expect this to happen (if at all)

Reply:
done